File: product-service/src/main/java/com/intexsoft/webshop/productservice/exception/conflict409/SuchResourceExistsException.java
package com.intexsoft.webshop.productservice.exception.conflict409;

import com.intexsoft.webshop.productservice.exception.Resource;
import com.intexsoft.webshop.productservice.exception.HttpStatusException;
import org.springframework.http.HttpStatus;

public abstract class SuchResourceExistsException extends HttpStatusException {
    private static final String MSG_PATTERN = "Such %s with name = %s already exists";

    public SuchResourceExistsException(Resource resource, String name) {
        super(String.format(MSG_PATTERN, resource.name(), name), HttpStatus.CONFLICT);
    }
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/exception/conflict/SuchResourceExistsException.kt
package com.intexsoft.webshop.productservicekt.exception.conflict

import com.intexsoft.webshop.productservicekt.exception.HttpStatusException
import com.intexsoft.webshop.productservicekt.exception.Resource
import org.springframework.http.HttpStatus

abstract class SuchResourceExistsException(resource: Resource, name: String) :
    HttpStatusException("Such ${resource.name} with name = $name already exists", HttpStatus.CONFLICT)